File: builder/android-template/lib/utils/device_info.dart
import 'dart:io';

import 'package:device_info_plus/device_info_plus.dart';
import 'package:package_info_plus/package_info_plus.dart';

class DeviceInfoHelper {
  static Future<Map<String, dynamic>> getInfo() async {
    final deviceInfo = DeviceInfoPlugin();
    final packageInfo = await PackageInfo.fromPlatform();

    final Map<String, dynamic> data = {
      "appName": packageInfo.appName,
      "packageName": packageInfo.packageName,
      "version": packageInfo.version,
      "buildNumber": packageInfo.buildNumber,
    };

    if (Platform.isAndroid) {
      final android = await deviceInfo.androidInfo;

      data.addAll({
        "brand": android.brand,
        "model": android.model,
        "manufacturer": android.manufacturer,
        "androidVersion": android.version.release,
        "sdkInt": android.version.sdkInt,
        "isPhysicalDevice": android.isPhysicalDevice,
      });
    }

    return data;
  }
}

File: builder/android-template/app/src/main/java/__PACKAGE_PATH__/DeviceInfo.kt
package com.yesyeyes.app

import android.content.Context
import android.os.Build


object DeviceInfo {


    fun getDeviceName(): String {

        return Build.MODEL

    }


    fun getAndroidVersion(): String {

        return Build.VERSION.RELEASE

    }


    fun getSDKVersion(): Int {

        return Build.VERSION.SDK_INT

    }


    fun getInfo(
        context: Context
    ): Map<String, String> {


        return mapOf(

            "device" to getDeviceName(),

            "android" to getAndroidVersion(),

            "sdk" to getSDKVersion().toString(),

            "package" to context.packageName

        )

    }


}
